Fix a bug with args

ramkumarvs opened the pull request:
Tried running the yb-ctl without any args and it errored, this fixes that issue.

Tested by `yb-ctl`

The error is due to the change in argparse behavior from 2 to 3.
https://stackoverflow.com/questions/22990977/why-does-this-argparse-code-behave-differently-between-python-2-and-3

Error message looks like

```
17:08 $ python3 bin/yb-ctl
Traceback (most recent call last):
  File "bin/yb-ctl", line 1969, in <module>
    control.run()
  File "bin/yb-ctl", line 1943, in run
    fallback_installation_dir=self.cluster_config.get("installation_dir"))
  File "bin/yb-ctl", line 716, in update_options_from_args
    if args.install_if_needed and not self.installation_dir:
AttributeError: 'Namespace' object has no attribute 'install_if_needed'
Viewing file /tmp/tmp8md0ltph:
^^^ Encountered errors ^^^
```

Commit:
Fix a bug with args

iSignal reviewed (approved):
looks good. wonder why travis ci is complaining?

iSignal commented:
looks like this may break the install if needed scenario for python 2? would be good to test it out.

iSignal committed:
Update the argparse setup

https://stackoverflow.com/questions/22990977/why-does-this-argparse-code-behave-differently-between-python-2-and-3 has some details of the changes in argparse from 2 to 3.

I made these change and checked basic usage against 2 and 3.

iSignal committed:
minor

extra line

iSignal committed:
remove whitespace

bmatican commented on bin/yb-ctl:
```diff
@@ -711,8 +711,7 @@ class ClusterOptions:
         # Automatic YugaByte DB installation
-        # -----------------------------------------------------------------------------------------
-
+        # -----------------------------------------------------------------------------------------  
```
> bin/yb-ctl:714:100: W291 trailing whitespace
can you undo this line? I think it broke our CI

iSignal replied:
done

bmatican commented on bin/yb-ctl:
```diff
@@ -821,7 +820,8 @@ class ClusterControl:
         self.parser = argparse.ArgumentParser()
-        self.subparsers = self.parser.add_subparsers()
+        self.subparsers = self.parser.add_subparsers(dest='command')
+        self.subparsers.required = True
```
shouldn't there be some default `help` that it defaults to if you don't specify this?

iSignal replied:
https://stackoverflow.com/questions/22990977/why-does-this-argparse-code-behave-differently-between-python-2-and-3

bmatican reviewed (changes requested):
Can you put a snippet of what it errored out with?

bmatican commented:
maybe also link to https://stackoverflow.com/questions/22990977/why-does-this-argparse-code-behave-differently-between-python-2-and-3 in the PR, as @iSignal is doing for `yb-docker-ctl`

iSignal committed:
trailing whitespace

bmatican reviewed (approved):
Maybe add the SO link to the description as well.
